From 6160625b0d44e86c8fc8605dc45ae0a5968b8dd9 Mon Sep 17 00:00:00 2001
From: Jakub Tyszkowski <jakub.tyszkowski@tieto.com>
Date: Tue, 2 Dec 2014 11:15:36 +0100
Subject: [PATCH 06/11] selinux: Add logwrapper security domain

Logwrapper needs its own domain so that proper domain transition rules
could be written for services launched with logwrapper from init.

Change-Id: I566660d826d7502d1dfa27f038d4442d49f019e8
---
 file_contexts | 2 +-
 logwrapper.te | 7 +++++++
 2 files changed, 8 insertions(+), 1 deletion(-)
 create mode 100644 logwrapper.te

diff --git a/file_contexts b/file_contexts
index 89bfa7f..d044470 100644
--- a/file_contexts
+++ b/file_contexts
@@ -173,7 +173,7 @@
 /system/bin/inputflinger u:object_r:inputflinger_exec:s0
 /system/bin/logd        u:object_r:logd_exec:s0
 /system/bin/uncrypt     u:object_r:uncrypt_exec:s0
-/system/bin/logwrapper  u:object_r:system_file:s0
+/system/bin/logwrapper  u:object_r:logwrapper_exec:s0
 /system/bin/vdc         u:object_r:vdc_exec:s0
 /system/bin/install-recovery.sh u:object_r:install_recovery_exec:s0
 /system/bin/dex2oat     u:object_r:dex2oat_exec:s0
diff --git a/logwrapper.te b/logwrapper.te
new file mode 100644
index 0000000..7895786
--- /dev/null
+++ b/logwrapper.te
@@ -0,0 +1,7 @@
+type logwrapper, domain;
+type logwrapper_exec, exec_type, file_type;
+
+# needs /system/bin/log access
+allow logwrapper devpts:chr_file rw_file_perms;
+
+init_daemon_domain(logwrapper)
-- 
2.9.3

